tests/analysis/test_analysis.py

```diff
@@ -55,31 +58,137 @@ def test_analysis(tmp_path: Path, ablator_results):
     )
     pass
 
+def test_name_remap(tmp_path: Path):
+    name_map = {'a': 'A', 'b': 'B'}
+    defaults = None
+    assert parse_name_remap(defaults, name_map) == name_map
 
-if __name__ == "__main__":
-    import shutil
-    from tests.ray_models.model import (
-        WORKING_DIR,
-        TestWrapper,
-        MyErrorCustomModel,
-        _make_config,
+    name_map = {'a': 'A', 'b': 'B'}
+    defaults = ['a', 'b', 'c', 'd']
+    expected_output = {'a': 'A', 'b': 'B', 'c': 'c', 'd': 'd'}
+    assert parse_name_remap(defaults, name_map) == expected_output
+
+    name_map = None
+    defaults = ['a', 'b', 'c']
+    assert parse_name_remap(defaults, name_map) == {'a': 'a', 'b': 'b', 'c': 'c'}
+
+    try:
+        parse_name_remap(None, None)
+    except NotImplementedError as err:
+        if str(err) != "`defaults` or `name_map` argument required.":
+            raise err
+
+def get_best(x: pd.DataFrame, task_type: str):
+    if task_type == "regression":
+        return x.sort_values("val_rmse", na_position="last").iloc[0]
+    else:
+        return x.sort_values("val_acc", na_position="first").iloc[-1]
+    
+def test_analysis_func(tmp_path: Path):
```

Comment:
I have added an asset because, The Results function requires `read_results ` to generate csv. The ablator_results just passes the result function. however, the dir is also needed.